### PR TITLE
Service Fabric add cert rollback on failure

### PR DESCRIPTION
**Pull request by a-santamaria**
## Description
When there is a failure thrown by VMSS backend validation, the VMSS model is still persisted so we need to trigger another update to revert the change. Added Rollback to fix this issue for Add-AzureRmServiceFabricApplicationCertificate and Add-AzureRmServiceFabricClusterCertificate. This fixes Azure/service-fabric-issues#932.

Also fixing adding cluster/application cert to only add to VM Scale Sets that correspond to the cluster by checking cluster id in Service Fabric extension and fixing some error messages, adding tests.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ServiceFabric/ServiceFabric/ServiceFabric.csproj [hunk 1]
```diff
@@ -16,4 +16,11 @@
     <PackageReference Include="Microsoft.Azure.Management.ServiceFabric" Version="1.1.1" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Update="Template\Linux\parameter.json" CopyToOutputDirectory="PreserveNewest" />
```
Comment (markcowl) [inline src/ServiceFabric/ServiceFabric/ServiceFabric.csproj]: Are these files required as part of the ServiceFabric module? If so, can we add a test that would prevent regressing on this in the future?

Comment (a-santamaria) [reply src/ServiceFabric/ServiceFabric/ServiceFabric.csproj]: yes, I'll add the test thanks.

Comment (a-santamaria) [reply src/ServiceFabric/ServiceFabric/ServiceFabric.csproj]: test added

File: src/ServiceFabric/ServiceFabric/ServiceFabric.csproj [hunk 2]
```diff
@@ -16,4 +16,11 @@
...
+    <None Update="Template\Linux\template.json" CopyToOutputDirectory="PreserveNewest" />
+    <None Update="Template\Windows\parameter.json" CopyToOutputDirectory="PreserveNewest" />
+    <None Update="Template\Windows\template.json" CopyToOutputDirectory="PreserveNewest" />
```
Comment (MiYanni) [inline src/ServiceFabric/ServiceFabric/ServiceFabric.csproj]: Please replace these 4 lines with this instead.
```xml
<None Update="Template\**\*.json" CopyToOutputDirectory="PreserveNewest" />
```

Comment (a-santamaria) [reply src/ServiceFabric/ServiceFabric/ServiceFabric.csproj]: done, thanks.

**Pull request merged**